core-bundle/src/Resources/contao/controllers/BackendMain.php

```diff
@@ -95,21 +94,12 @@ public function __construct()
 				throw new \RuntimeException('The request stack did not contain a request');
 			}
 
-			$objJwtManager = $objRequest->attributes->get(JwtManager::REQUEST_ATTRIBUTE);
-			$script = Input::get('enable') ? '/preview.php' : '';
-
-			if (!$objJwtManager instanceof JwtManager)
-			{
-				if (($qs = $objRequest->getQueryString()) !== null)
-				{
-					$qs = '?' . $qs;
-				}
-
-				$this->redirect($script . $objRequest->getPathInfo() . $qs);
-			}
+			// This will throw an exception if the JwtManager does not exist (not a Managed Edition).
+			// As we do not show the debug button in that case, this should never be reached.
+			$objJwtManager = System::getContainer()->get('contao_manager.jwt_manager');
 
 			$strReferer = Input::get('referer') ? '?' . base64_decode(Input::get('referer', true)) : '';
-			$objResponse = new RedirectResponse($script . $objRequest->getPathInfo() . $strReferer);
+			$objResponse = new RedirectResponse($objRequest->getPathInfo() . $strReferer);
```

Comment:
So this has two implications:

1. If you do not have a ME, you cannot have the debug mode enabled without implementing the `contao_manager.jwt_manager` service yourself? Maybe we could improve that?

2. If you're not logged in, have a cache entry, and enable the debug mode, you'll never see it...

Reply:
1. If you do not have a managed edition, I don‘t think a back end user should decide on debug mode. You would run a manual AppKernel and use environment variables. 

2. that could be correct, for the front end. But if you use the preview button, everything works. And you also have a JWT cookie, so your cache will be bypassed in most cases.

Reply:
1. Fair point.
2. You have a cookie, so it's bypassed. True.